### PR TITLE
feat(adapter-next): add `<SliceSimulator>`

### PR DESCRIPTION
**Pull request by angeloashmore**
## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR adds `<SliceSimulator>` to `@slicemachine/adapter-next`. This change replaces the `@prismicio/slice-simulator-react` package. Next.js developers will no longer need to install a separate Slice Simulator package as it will already be available from the adapter.

## Checklist:

- [ ] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

🐮

### CODE REVIEW

File: packages/adapter-next/siroc.config.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { defineSirocConfig } from "siroc";
 export default defineSirocConfig({
 	rollup: {
 		output: {
+			exports: "named",
```
Comment (lihbr) [inline packages/adapter-next/siroc.config.ts:6]: Just out of curiosity / for my learning's sake, but why is this needed here? I guess it has to do with the named exports and export default in index 🤔

**Review by lihbr — APPROVED** [review]
LGTM

File: packages/adapter-next/siroc.config.ts (hunk 1 above)
Comment (angeloashmore) [reply packages/adapter-next/siroc.config.ts:6]: Yes, exactly for that reason. Rollup printed the following message when a default export and named exports were mixed in the same file.

```
Entry module "src/index.ts" is using named and default exports together. Consumers of your bundle will have to use `chunk["default"]` to access the default export, which may not be what you want. Use `output.exports: "named"` to disable this warning
```

Since we read `plugin.default` when importing the plugin, I think it's okay that we export it as `default`.
